File: platform/frontend/src/types/cxo.ts
// CXO dashboard types — matches backend schemas/cxo.py

export interface VelocityPoint {
  sprint: string | null;
  committed: number;
  completed: number;
  end_date: string | null;
}

export interface BurndownPoint {
  date: string;
  remaining: number;
}

export interface Burndown {
  sprint: string | null;
  total: number;
  series: BurndownPoint[];
}

export interface AssigneeTickets {
  assignee: string;
  todo: number;
  in_progress: number;
}

export interface CycleTimePoint {
  key: string;
  days: number;
}

export interface IssueTypeCount {
  type: string;
  count: number;
}

export interface ThroughputPoint {
  date: string;
  done: number;
}

export interface ProjectSummary {
  key: string;
  total: number;
  done: number;
  in_progress: number;
  todo: number;
  completion_pct: number;
}

export interface ProjectMetrics {
  summary: ProjectSummary;
  velocity: VelocityPoint[];
  burndown: Burndown;
  assignees: AssigneeTickets[];
  cycle_time: CycleTimePoint[];
  issue_types: IssueTypeCount[];
  throughput: ThroughputPoint[];
  cached: boolean;
}

export interface Portfolio {
  projects: ProjectSummary[];
}

File: platform/backend/app/schemas/cxo.py
"""Pydantic schemas for CXO dashboard — PRJ0-21."""

from __future__ import annotations
from typing import Any
from pydantic import BaseModel


class VelocityPoint(BaseModel):
    sprint: str | None
    committed: float
    completed: float
    end_date: str | None


class BurndownPoint(BaseModel):
    date: str
    remaining: float


class Burndown(BaseModel):
    sprint: str | None
    total: float
    series: list[BurndownPoint]


class AssigneeTickets(BaseModel):
    assignee: str
    todo: int
    in_progress: int


class CycleTimePoint(BaseModel):
    key: str
    days: float


class IssueTypeCount(BaseModel):
    type: str
    count: int


class ThroughputPoint(BaseModel):
    date: str
    done: int


class ProjectSummary(BaseModel):
    key: str
    total: int
    done: int
    in_progress: int
    todo: int
    completion_pct: float


class ProjectMetrics(BaseModel):
    summary: ProjectSummary
    velocity: list[VelocityPoint]
    burndown: Burndown
    assignees: list[AssigneeTickets]
    cycle_time: list[CycleTimePoint]
    issue_types: list[IssueTypeCount]
    throughput: list[ThroughputPoint]
    cached: bool = False


class Portfolio(BaseModel):
    projects: list[ProjectSummary]
